src/main/java/me/dinowernli/grpc/polyglot/io/MessageWriter.java

```diff
@@ -22,34 +23,43 @@
   private static final Logger logger = LoggerFactory.getLogger(MessageWriter.class);
 
   /** Used to separate the individual plaintext json proto messages. */
-  private static final String MESSAGE_SEPARATOR = "\n\n";
+  private final String messageSeparator;
 
   private final JsonFormat.Printer jsonPrinter;
   private final Output output;
+  private boolean firstMessagePrinted = false;
 
   /**
    * Creates a new {@link MessageWriter} which writes the messages it sees to the supplied
    * {@link Output}.
    */
-  public static <T extends Message> MessageWriter<T> create(Output output) {
-    return new MessageWriter<T>(JsonFormat.printer(), output);
+  public static <T extends Message> MessageWriter<T> create(Output output, String messageSeparator) {
```

Comment:
Let's have two factory methods, one called create(), and the other called createWithSeparator().

That way, there is a sensible default and we don't need the ServiceCall class to understand how these separators work.